### PR TITLE
Adding basic auth to catalog-web

### PR DESCRIPTION
**Pull request by btylerburton**
Relates to:
  - GSA/data.gov#3910

### CODE REVIEW

File: proxy/.profile [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+export SPACE_NAME=$(echo $VCAP_APPLICATION | jq -r '.space_name')
+
+echo "Setting up proxy in $APP_NAME on $SPACE_NAME"
+
+# only add auth to staging environment / otherwise turn it off
+if [[ $APP_NAME = "catalog-proxy" ]] && \
+[[ $SPACE_NAME = "staging" ]]
+then
+    echo "Setting proxy basic auth..."
+    echo "$(vcap_get_service secrets .credentials.PROXY_AUTH.USERNAME):$(vcap_get_service secrets .credentials.PROXY_AUTH.PASSWORD)" > ${HOME}/etc/nginx/.htpasswd
```
Comment (nickumia-reisys) [inline proxy/.profile:24]: What happens if there is a parse error here?

Comment (btylerburton) [reply proxy/.profile:24]: Good question. Are there any examples of parsing with a fallback in the repo?

**Comment by btylerburton** [comment]
@jbrown-xentity what would be the desired action if we hit an error?

**Comment by btylerburton** [comment]
Gave this some thought over the weekend and I think if the values are missing we don't want to skip adding auth, or add auth in a broken state, so might be best to fail the build. 

Will chat with @FuhuXia and @Jin-Sun-tts about this and get their input, then make changes accordingly.

**Comment by FuhuXia** [comment]
Can we see it in action on `development` space?

**Review by FuhuXia — APPROVED** [review]

**Pull request merged**